From 6310769c3c022300dfd9903ff822ac43fba6446f Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <esmil@mailme.dk>
Date: Fri, 19 Sep 2014 19:43:10 +0200
Subject: [PATCH ] string.h: add strndupa

---
 include/string.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/string.h b/include/string.h
index 795a2abc..268cddde 100644
--- a/include/string.h
+++ b/include/string.h
@@ -87,6 +87,13 @@ void explicit_bzero (void *, size_t);
 
 #ifdef _GNU_SOURCE
 #define	strdupa(x)	strcpy(alloca(strlen(x)+1),x)
+#define	strndupa(x, l)	({ \
+	const char *__old = (x); \
+	size_t __len = strnlen(__old, (l)); \
+	char *__new = (char *)alloca(__len + 1); \
+	__new[__len] = '\0'; \
+	(char *)memcpy(__new, __old, __len); \
+})
 int strverscmp (const char *, const char *);
 char *strchrnul(const char *, int);
 char *strcasestr(const char *, const char *);
--
2.11.0
