diff --git a/deps/jemalloc/configure.ac b/deps/jemalloc/configure.ac
index 7a1290e0..877423fe 100644
--- a/deps/jemalloc/configure.ac
+++ b/deps/jemalloc/configure.ac
@@ -1107,6 +1107,7 @@ if test "x$LG_PAGE" = "xdetect"; then
 ]])],
                              [je_cv_lg_page=`cat conftest.out`],
                              [je_cv_lg_page=undefined],
+                             [je_cv_static_page_shift=12],
                              [je_cv_lg_page=12]))
 fi
 if test "x${je_cv_lg_page}" != "x" ; then
diff --git a/deps/jemalloc/include/jemalloc/internal/atomic.h b/deps/jemalloc/include/jemalloc/internal/atomic.h
index a9aad35d..adae7c19 100644
--- a/deps/jemalloc/include/jemalloc/internal/atomic.h
+++ b/deps/jemalloc/include/jemalloc/internal/atomic.h
@@ -467,8 +467,7 @@ atomic_write_uint32(uint32_t *p, uint32_t x)
 
 	InterlockedExchange(p, x);
 }
-#elif (defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) || \
- defined(JE_FORCE_SYNC_COMPARE_AND_SWAP_4))
+#elif (defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) || defined(__arm__))
 JEMALLOC_INLINE uint32_t
 atomic_add_uint32(uint32_t *p, uint32_t x)
 {
